HIPs/incubation/xkey.md

```diff
@@ -0,0 +1,136 @@
+This initiative aims to establish an ecosystem for threshold signing schemes (TSS) 
+and multi-party computations (MPC) in general.
+
+# Context
+Signing schemes are a crucial part of our daily technology. They power systems like 
+TLS, code signing, wallets, blockchains and others.
+
+In basic cryptography, signing schemes use a pair of keys: a private key and a public key. 
+The private key signs messages, and the public key verifies the signature.
+
+Keeping the private key secret is essential. If it's leaked, it can lead to serious 
+problems, like someone stealing all the funds in a crypto wallet. Losing the private key 
+is also a big issue, as it can result in permanent loss of access to funds.
+
+To manage these risks, we can use Threshold Signing Schemes (TSS). TSS is an advanced 
+method that uses multiple private key shares among multiple participants. In a t-out-of-n 
+scheme, `n` signers share the key, and any `t` of them can work together to sign a message. 
+This setup offers unique security benefits: an attacker can compromise up to `t-1` signers 
+without learning anything about the private key. Also, up to `n-t` signers can fail without 
+affecting service availability.
+
+However, TSS protocols are complex. The cryptographic principles behind them are hard to 
+understand, and their implementation adds more layers of complexity. TSS protocols are 
+usually multi-round and require synchronous communication with authenticated and encrypted 
+channels. Their state progresses gradually as they receive messages. Therefore, implementing 
+TSS involves intricate boilerplate code, making it difficult to audit.
+
+# Dependent Projects
+None
+
+# Motivation
+When we implemented a TSS ECDSA protocol, we noticed a lack of tools and frameworks available 
+in the community. Most TSS implementations either rewrite a state machine from scratch to 
+drive protocol execution or offer low-level functions, requiring manual protocol handling. 
+This results in boilerplate code that is difficult to read and maintain.
+
+Our goal is to build an ecosystem with:
+* A framework for writing MPC protocols, reducing boilerplate and making the code easy to
+  write and read.
+* A library for generic elliptic curve cryptography, enabling seamless curve switching and
+  secure arithmetic operations, while managing small subgroups and memory security.
+* Common primitives and ZK proofs like Schnorr Proof of Knowledge, unambiguous hashing,
+  secret sharing, and more.
+* Applied libraries like fast Paillier encryption, using CRT to speed up (de)encryption
+  and homomorphic operations, among others.
+
+The ultimate goal of this proposal is to make developing and maintaining TSS/MPC protocols 
+easier. We want to improve their reliability, readability, and security because we believe 
+these are the safest and most feature-rich ways to protect and move digital assets.
+
+# Status
+Pending proposal
+
+# Solution
+We developed following crates in Rust:
```

Comment:
any plans on supporting Golang?

Reply:
even wrapper would be very helpful.

Reply:
Not with our resources, but if somebody from community comes with such contribution (and strong intention to keep developing and maintaining it), I think we'd be happy to accept it.